app/uk/gov/hmrc/vatsubscription/models/InflightReturnPeriod.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.vatsubscription.models
+
+import play.api.Logger
+import play.api.libs.json._
+import uk.gov.hmrc.vatsubscription.config.AppConfig
+
+sealed trait InflightReturnPeriod {
+  def returnPeriod: String
+}
+
+case object MAInflightReturnPeriod extends InflightReturnPeriod {
+  override val returnPeriod: String = "MA"
+}
+
+case object MBInflightReturnPeriod extends InflightReturnPeriod {
+  override val returnPeriod: String = "MB"
+}
+
+case object MCInflightReturnPeriod extends InflightReturnPeriod {
+  override val returnPeriod: String = "MC"
+}
+
+case object MMInflightReturnPeriod extends InflightReturnPeriod {
+  override val returnPeriod: String = "MM"
+}
+
+case object InvalidInflightReturnPeriod extends InflightReturnPeriod {
+  override val returnPeriod: String = "XX"
+}
+
+object InflightReturnPeriod {
+
+  def apply(period: String): InflightReturnPeriod = period match {
+    case MAInflightReturnPeriod.returnPeriod => MAInflightReturnPeriod
+    case MBInflightReturnPeriod.returnPeriod => MBInflightReturnPeriod
+    case MCInflightReturnPeriod.returnPeriod => MCInflightReturnPeriod
+    case MMInflightReturnPeriod.returnPeriod => MMInflightReturnPeriod
+    case _ =>
+      Logger.warn(s"[ReturnPeriod][apply] Invalid Return Period: '$period'")
+      InvalidInflightReturnPeriod
+  }
+
+  def unapply(arg: InflightReturnPeriod): String = arg.returnPeriod
+
+  implicit val readsV3_2_1: Reads[InflightReturnPeriod] = {
```

Comment:
These probably shouldn't be implicit anymore as we want them to be explicitly used. Also, I suspect if something tried to implicitly use it it would complain that there are two implicit Reads[InflightReturnPeriod] in scope and wouldn't know which to resolve to.

Reply:
Note, this change needs to be made throughout